src/iqm/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -28,7 +28,7 @@
 from qiskit_aer.noise import NoiseModel
 from qiskit_aer.noise.errors import depolarizing_error, thermal_relaxation_error
 
-from iqm.iqm_client import QuantumArchitectureSpecification
+from iqm.iqm_client import DynamicQuantumArchitecture, QuantumArchitectureSpecification
```

Comment:
The reasons why I still used QuantumArchitectureSpecification for FakeBackend are:

* Conceptually, FakeBackend does not have a calibration set, while a DynamicQuantumArchitecture always corresponds to a calibration set.

* More convenient to create the FakeBackends, since there is no need to add fake gate implementation info and fake calset ids when defining them. It makes their definitions cleaner since there is not this extra info which would be unused anyway.

* Keeping it using QuantumArchitectureSpecification requires less code changes, less changes to review.

* In the follow-up issue SW-707 where we change how static architectures work, there is a note about the possibility to use the static/idealistic arhcitecture to create the simulated backends. If we change them now to use DQA, we might need to change again to use the new SQA (so we change `old SQA -> DQA -> new SQA`), while possibly it would be enough to just do the change then (`old SQA -> new SQA`).

Because of these reasons, I would still keep FakeBackend using QuantumArchitectureSpecification, and revisit it in the follow-up issue.

Reply:
Not allowing SQA for IQMBackendBase makes the library unusable for devices that don't support DQA yet. So that's not good. What I'm saying is that the IQMFakeBackend should support both as well, because we need have the SQA available for some of these devices right now, and we want to use it for creating an idealistic fake backend. But sometimes you want an accurate fake backend, for which you also need the DQA.

Reply:
I see, I reverted the commit which removed SQA support from IQMBackendBase. But I would still not add DQA support to IQMFakeBackend, because we currently only support hardcoded FakeBackends, and the above reasons are valid considering. I think IQMFakeBackend support for DQA should be done in the issue which actually adds support for IQMFakeBackends using real calibration data.